subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyTaskIntegrationSpec.groovy

```diff
@@ -84,6 +82,21 @@ class CopyTaskIntegrationSpec extends AbstractIntegrationSpec {
             'c.txt',
             'sub/empty'
         )
+
+        when:
+        file("files/sub/empty2").createDir()
+        run 'copy'
+
+        then:
+        executedAndNotSkipped(":copy")
+        file('dest').assertHasDescendants(
+            'sub/a.txt',
+            'sub/dir/b.txt',
+            'sub/d.txt',
+            'c.txt',
+            'sub/empty',
+            'sub/empty2'
+        )
```

Comment:
Why add this? We already test empty directory handling with `sub/empty`, no?

Reply:
We don't. This is about changing an empty directory (by adding one), not about skipping a task because there are only empty directories.

Reply:
I see. It would be good to clarify this, though. This test looks big, so breaking out another one with a separate name could be good. Or adding some comments here about the difference between `sub/empty` and `sub/empty2`. Another option would be to give more descriptive names to these directories, related to their roles.

Reply:
I split the test.